### PR TITLE
flattening: pass provider cluster roles through from helm chart

### PR DESCRIPTION
**Pull request by tfussell**
Towards https://github.com/giantswarm/giantswarm/issues/7820

KVM TC nodes run in pods on the control plane. This operator creates unique service accounts for each TC, but the cluster role bound to the SA is shared between the operator and the pods and is created by the helm chart. It needs to be passed through from the helm chart to the `clusterrolebinding` resource so that pods use the cluster role corresponding to the operator that is reconciling it.

### CODE REVIEW

File: helm/kvm-operator/templates/configmap.yaml [hunk 1]
```diff
@@ -14,6 +14,8 @@ data:
     service:
       crd:
         labelSelector: '{{ .Values.Installation.V1.GiantSwarm.KVMOperator.CRD.LabelSelector }}'
+      podSecurityPolicy:
+        name: {{ tpl .Values.resource.psp.name . }}
```
Comment (tfussell) [inline helm/kvm-operator/templates/configmap.yaml]: Not sure if this needs to be quoted like `labelSelector` above.

Comment (tfussell) [reply helm/kvm-operator/templates/configmap.yaml]: I think this is always going to be a valid YAML string so it's not necessary to quote.

File: service/controller/resource/clusterrolebinding/resource.go [hunk 2]
```diff
@@ -18,26 +18,35 @@ const (
...
+	}
+	if config.PodSecurityPolicyName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.PodSecurityPolicyName must not be empty", config)
 	}
 
 	newService := &Resource{
 		k8sClient: config.K8sClient,
 		logger:    config.Logger,
+
+		podSecurityPolicyName: config.PodSecurityPolicyName,
```
Comment (kopiczko) [inline service/controller/resource/clusterrolebinding/resource.go]: Did you test upgrades? Is the PSP name changed during the upgrade and everything is fine?

File: service/service.go [hunk 3]
```diff
@@ -136,9 +136,10 @@ func New(config Config) (*Service, error) {
 			Logger:        config.Logger,
 			TenantCluster: tenantCluster,
 
-			CRDLabelSelector:   config.Viper.GetString(config.Flag.Service.CRD.LabelSelector),
-			GuestUpdateEnabled: config.Viper.GetBool(config.Flag.Service.Tenant.Update.Enabled),
-			ProjectName:        project.Name(),
+			CRDLabelSelector:      config.Viper.GetString(config.Flag.Service.CRD.LabelSelector),
+			GuestUpdateEnabled:    config.Viper.GetBool(config.Flag.Service.Tenant.Update.Enabled),
+			PodSecurityPolicyName: config.Viper.GetString(config.Flag.Service.PodSecurityPolicy.Name),
+			ProjectName:           project.Name(),
```
Comment (xh3b4sd) [inline service/service.go]: Just for the future. With the `pkg/project` package we do not need to inject that anywhere anymore. We can simply use the package where needed. The project package should simply be imported directly where necessary.

File: service/controller/resource/clusterrolebinding/resource.go (hunk 2 above)
Comment (tfussell) [reply service/controller/resource/clusterrolebinding/resource.go]: Good point. I will investigate.

File: service/service.go (hunk 3 above)
Comment (tfussell) [reply service/service.go]: Thanks for mentioning. I might have missed that during flattening. I'll be sure to resolve in a separate PR.

File: service/controller/resource/clusterrolebinding/resource.go (hunk 2 above)
Comment (tfussell) [reply service/controller/resource/clusterrolebinding/resource.go]: The previous hardcoded PSP is `kvm-operator-psp` which should still be created by the `thiccc` operator. In the case of a flattened operator, the required PSP is created when the operator is deployed via helm so it must exist by the time a cluster is being upgraded as far as I can tell. I will deploy this branch to make sure that my thinking is correct here.

Comment (kopiczko) [reply service/controller/resource/clusterrolebinding/resource.go]: > the required PSP is created when the operator is deployed via helm

If this is in `resource` package then this code is executed only during CR reconciliation. So surely not right after deployment. I'm not clear on what the PSP is used for though.

Comment (tfussell) [reply service/controller/resource/clusterrolebinding/resource.go]: I don't fully understand `kvm-operator` myself, but I believe this PSP is used for the pods on the control plane that run VMs that run the TC nodes. So basically the operator pod and the node pods share the same PSP.

**Review by MarcelMue — APPROVED** [review]
LGTM if tested successfully.

**Comment by tfussell** [comment]
Tested on `geckon`. Cluster creation looks good. Upgrade from 11.0.0 to 11.1.0 worked without any problems. No errors in kvm-operator logs.

File: service/controller/resource/clusterrolebinding/resource.go (hunk 2 above)
Comment (tfussell) [reply service/controller/resource/clusterrolebinding/resource.go]: Good call on testing upgrades. Sussed out a few problems in the process.

**Review by MarcelMue — APPROVED** [review]
I already approved but let's get this merged :+1:

File: service/service.go (hunk 3 above)
Comment (tfussell) [reply service/service.go]: Started here https://github.com/giantswarm/kvm-operator/pull/824

**Pull request merged**